### ISSUE TITLE
Eliminate board_initialise()

### ISSUE THREAD

**Issue by hadley** [rstudio/pins#382]
```R
board_register_local <- function(name, cache, ...) {
  board_register("local", name, cache, ...)
  # which eventually calls board_initialize.local()
}

# ->

board_register_local <- function(name = "local",
                                 cache = board_cache_path(),
                                 ...) {
  
  board <- board_local(name = name, cache = cache, ...)
  board_register("local", name = name, cache = cache)
}

#' @export
board_local <- function(name = "local", 
                        cache = board_cache_path(), 
                        versions = FALSE) {
  
  dir.create(cache, recursive = TRUE, showWarnings = FALSE)
  new_board("local", name = name, cache = cache, versions = versions)
}
```

Will need new name for a `board_register()` that only registers (not creates) the board.

**Comment by hadley** [comment]
For now, drop code to recreate pins since analysis suggests it's either dangerous or ineffective.

**Comment by hadley** [comment]
Need a basic print method

**Comment by hadley** [comment]
Fixed in #389
